.vitepress/theme/components/sidebar/SideBarLink.ts

```diff
@@ -14,21 +15,32 @@ export const SideBarLink: FunctionalComponent<{
   const route = useRoute()
   const site = useSiteData()
 
-  const headers = route.data.headers
+  const headers = route.data?.headers
```

Comment:
`route.data` is always have value, if not it will throw error.

Reply:
It should go to 404 page that you just added.

Reply:
@hannoeru There is some weird behaviour when you visit a missing page, `data` is null so any reference to it breaks the page.

See the console and visit a 404 page

Reply:
Looks like vitepress couldn't handle dynamic route, so my suggestion is delete 404 page & redirect, let netlify handle it.

Reply:
I'll just revert all the stuff I did around the redirects & 404 for now